src/ompl/tools/lightning/src/Lightning.cpp

```diff
@@ -0,0 +1,464 @@
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the JSK, The University of Tokyo nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dave Coleman */
+
+
+#include "ompl/tools/lightning/Lightning.h"
+#include "ompl/tools/lightning/LightningDB.h"
+
+namespace og = ompl::geometric;
+namespace ob = ompl::base;
+namespace ot = ompl::tools;
+
+ompl::tools::Lightning::Lightning(const base::SpaceInformationPtr &si) :
+    ompl::tools::ExperienceSetup(si)
+{
+    initialize();
+}
+
+ompl::tools::Lightning::Lightning(const base::StateSpacePtr &space) :
+    ompl::tools::ExperienceSetup(space)
+{
+    initialize();
+}
+
+void ompl::tools::Lightning::initialize()
+{
+    recallEnabled_ = true;
+    scratchEnabled_ = true;
+    filePath_ = "unloaded";
+
+    // Load dynamic time warp
+    dtw_.reset(new ot::DynamicTimeWarp(si_));
+
+    // Load the experience database
+    experienceDB_.reset(new ompl::tools::LightningDB(si_->getStateSpace()));
+
+    // Load the Retrieve repair database. We do it here so that setRepairPlanner() works
+    rrPlanner_ = ob::PlannerPtr(new og::LRR(si_, experienceDB_));
+
+    OMPL_INFORM("Lightning Framework initialized.");
+}
+
+bool ompl::tools::Lightning::setFile(const std::string &databaseName, const std::string &databaseDirectory)
+{
+    std::string fileName = "lightning_" + databaseName;
+    return ompl::tools::ExperienceSetup::getFilePath(fileName, databaseDirectory);
+}
+
+void ompl::tools::Lightning::setup(void)
+{
+    if (!configured_ || !si_->isSetup() || !planner_->isSetup() || !rrPlanner_->isSetup() )
+    {
+        OMPL_ERROR("Setting up Lightning Framework");
```

Comment:
Should not be, fixed (sometimes I use red to grab my attention in a sea of debug output)